src/metabase/api/setup.clj

```diff
@@ -39,6 +39,9 @@
     "Token does not match the setup token."))
 
 (defn- setup-create-user! [{:keys [email first-name last-name password]}]
+  (when-not config/is-test?
+    ;; many tests use /api/setup to setup multiple users
+    (assert (not (setup/has-user-setup)) "setup-create-user! can only be called when no users have been setup."))
```

Comment:
1. This should be i18n'ed since there's a chance it can be hit by someone who isn't a actively hacking on Metabase
2. This should throw an `ExceptionInfo` with a `:status-code`. Otherwise I think this would probably end up returning a 500 which is not appropriate here
```suggestion
    (when-not (setup/has-user-setup)
      (throw (ex-info (tru "Setup endpoints can not be used after a user has been created.") {:status-code 400})))
```

Reply:
I think having that `config/is-test?` flag is actually not the best way to go about this since it effectively makes this logic impossible to test. I think adding a dynamic variable like `*disallow-setup-api-calls-after-first-user-is-created*` (default `false`) and then binding that to `true` in relevant tests is better than just a blanket disabling of this logic in tests

Reply:
Makes sense. I have added `*allow-setup-api-calls-after-first-user-is-created*` and am working on testing it